src/main/java/Duke.java

```diff
@@ -1,10 +1,245 @@
+     */
+    private static void printList() {
+        for (int i = 0; i < listIndexTracker; i++) {
+            System.out.format(TAB + "% 3d.", i + 1);
+            System.out.println(list[i]);
+        }
+    }
+
+    /**
+     * This function sets done to true, stopping while loop and signifying the end of program.
+     */
+    private static void setDone() {
+        done = true;
+    }
+
+    /**
+     * This function check if done statement and index in done statement is valid.
+     * @param input input given by the user.
+     * @return returns the validity of the done input.
+     */
+    private static boolean isDoneInput(String input) {
+        boolean validDoneStatement = Pattern.matches("^done \\d+$", input.toLowerCase());
+        if (!validDoneStatement) {
+            return false;
+        }
+        int numberOnly = Integer.parseInt(input.replaceAll("[^0-9]", ""));
+        return numberOnly <= listIndexTracker && numberOnly < 100;
+    }
+
+    /**
+     * This function retrieves the index given in done statement.
+     * @param input input given by the user.
+     * @return returns the index given in done statement
+     */
+    private static int getDoneIndex(String input) {
+        String[] parts = input.split(" ");
+        return Integer.parseInt(parts[1]) - 1;
+    }
+
+    /**
+     * This function sets error as the given boolean argument.
+     * @param bool boolean to set error as.
+     */
+    private static void setError(boolean bool) {
+        error = bool;
+    }
+
+    /**
+     * This function uses regex to check if to do statement is valid.
+     * @param input input given by the user.
+     * @return returns validity of the to do statement.
+     */
+    private static boolean isValidTodo(String input) {
+        return Pattern.matches("todo [a-z0-9\\s]+", input.toLowerCase());
+    }
+
+    /**
+     * This function uses regex to check if deadline statement is valid.
+     * @param input input given by the user.
+     * @return returns validity of the deadline statement.
+     */
+    private static boolean isValidDeadline(String input) {
+        return Pattern.matches("deadline [a-z0-9\\s]+\\b /by .+", input.toLowerCase());
+    }
+
+    /**
+     * This function uses regex to check if event statement is valid.
+     * @param input input given by the user.
+     * @return returns validity of the event statement.
+     */
+    private static boolean isValidEvent(String input) {
+        return Pattern.matches("event [a-z0-9\\s]+\\b /at .+", input.toLowerCase());
+    }
+
+    /**
+     * This function gets the parameters for the Task subclasses by slicing input.
+     * @param parameters Array of string of fixed size 2 to store parameters for Task subclasses.
+     * @param input User's input into command line.
+     * @param taskType String that is pre-identified (DEADLINE/EVENT/TODO)
+     */
+    private static void getParameters(String[] parameters, String input, String taskType) {
```

Comment:
Consider converting the `taskType` into an enumeration instead.
https://nus-cs2113-ay2122s1.github.io/website/schedule/week4/topics.html#w4-4-oop-java-enumerations